{{cookiecutter.project_slug}}/tsconfig.json

```
@@ -0,0 +1,70 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    "incremental": true,                   /* Enable incremental compilation */
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
```

We have `module` specified in the package.json https://github.com/netlify/node-template/blob/main/package.json#L6

So we should compile to ES Modules

 ```suggestion
    "module": "ESNext",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
```

---

I am wondering whether `nodenext` should be used instead? Note: unfortunately, this is still in experimental mode.
When I tried using `"module": "esnext"` in ZISI when migrating to ES modules, this did not quite work, as I think that mode is meant for browsers, while `nodenext` is meant for pure ES modules in Node.js.

---

Doesn't `ESNext` mean features that are not on the spec yet?

---

In principle, yes. [The documentation mentions](https://www.typescriptlang.org/tsconfig#module) that `es2015` already outputs ES modules, but `es2020` adds dynamic `import` and `import.meta`, and `nodenext` adds pure ES modules support (`"type": "module" in `package.json`).

---

So it feels like `es2020` would be a sensible choice here?

---

yup, agreeing with `es2020`!

```suggestion
    "module": "es2020",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
```